### ISSUE TITLE
MkvToolNix at Windows breaks FileHippoApp Manager

### ISSUE THREAD

**Issue by AnselmD** [mbunkus/mkvtoolnix#1490]
Hi,

this is more an information and less a bug report.

The installation of the MKVToolNix (8.5.1) at my Windows 7 system breaks the FileHippoApp Manager, which informs about new software versions of various programs.

You can find my comments at http://filehippo.com/de/download_app_manager/comments/

**Comment by mbunkus** [comment]
That's a bug in FileHippoApp.

**Comment by AnselmD** [comment]
Yes, only for interest. You are using NSIS? Did you change anthing from the prevoius version to the 8.5.1 in the installation process?

**Comment by AnselmD** [comment]
What did you change from 8.3.0 to 8.4.0? Do you use another compression algorithm with NSIS? Or did you change the version?
The files size of 8.3.0 is about 19.1 MB and the file size of 8.4.0 (which is the first, which does not work) is about 13.2 MB.

**Comment by mbunkus** [comment]
In 8.4.0 wxWidgets support was dropped and therefore mmg was removed. The registry key where the installer put its settings was changed, too, as the prior key contained the word `mmg`.

If you want to see the actual differences then look at the output of `git diff release-8.3.0..release-8.4.0 -- installer/mkvtoolnix.nsi`.

**Comment by mbunkus** [comment]
Oh and I do receive notifications for anything you post here even if it's closed. If I don't answer quickly then I don't have the time to.

**Comment by AnselmD** [comment]
Thank you very much

**Comment by mbunkus** [comment]
Just to clarify a few points:

- MKVToolNix does not interface with FileHippo at all. I don't know how FileHippo works, but I guess it searches the registry for known keys and retrieves version information either from those keys or from certain applications (.exe) it assumes are installed.
- The old GUI (mkvmerge GUI, ``mmg.exe``) was removed in 8.4.0. Therefore ``mmg.exe`` is not installed anymore. If FileHippo assumes ``mmg.exe`` is present and tries to retrieve its version by reading from it then that's their bug.
- Similar things can be said for the registry key ``Software\Microsoft\Windows\CurrentVersion\App Paths\mmg.exe``. As ``mmg.exe`` itself isn't present anymore my installer now uses ``Software\Microsoft\Windows\CurrentVersion\App Paths\mkvtoolnix-gui.exe``. Again, if FileHippo has a problem with this registry key (or the old one missing) then that's their bug.

I'm definitely responsible for bugs in my software. I'm not reponsible for bugs in other people's software, and I refuse to implement workarounds for such kinds of bugs as implementing workarounds doesn't scale, can lead to problems with third-party software that actually work fine with the current MKVToolNix, and on top of that implementing workarounds has a cost both in time to implement and test them intially as well as maintaining them in the future.

**Comment by AnselmD** [comment]
Obviously it is a bug in FileHippo Update Manager and not in MKVToolNix. 

As a workarround somebody at the comments page of FileHippo Update Manager recommands to use the portable version of Mkvtoolnix. My workarround is to use MKVToolNix for Linux.

Nevertheless the developer of File Hippo Update Manger should have enough information now to fix the bug and they should improve the error handling, leading to misunderstandable errors. 

I wrote an email to them, three weeks ago, but did not get an answer. So users who like to have a fix, should also write an email, the address is mentioned at the FAQ page:
http://filehippo.com/de/info/appmanager/

So you are not responsable for the difficulties, but you shoud be aware of it. Do not waste your time with it.

**Comment by AnselmD** [comment]
BTW: The comment page of File Hippo Update Manager is closed. "Comments for this thread are now closed". My intention was to encourage the affected users to write an email to the developer. (http://filehippo.com/de/info/appmanager/)

Nevertheless, do not waste your time with it. And thanks again for your attention.

**Comment by mbunkus** [comment]
@AnselmD I posted the clarification not so much for your benefit but for the other folks who haven't looked into this as much yet; e.g. the guy reporting #1515.

Judging from a little digging I did yesterday the FileHippo developers don't seem to be too keen on communicating with their users; neither don't they react quickly to obvious bugs. A shame, really, but you're right, I won't bother with contacting them.
